src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -201,6 +232,45 @@ private bool Deserialize(Stream stream, out GraphQLRequest single, out GraphQLRe
                 }
             }
         }
+#else
+        private async Task<(bool wasDeserialized, GraphQLRequest single, GraphQLRequest[] batch)> DeserializeAsync(Stream stream)
+        {
+            using var ms = new MemoryStream();
+            await stream.CopyToAsync(ms).ConfigureAwait(false);
+            var jsonBytes = ms.ToArray();
+
+            JsonTokenType tokenType;
+            try
+            {
+                tokenType = GetTokenType(jsonBytes);
+            }
+            catch (Exception)
+            {
+                return (false, null, null); // BadRequest
+            }
+
+            switch (tokenType)
+            {
+                case JsonTokenType.StartObject:
+                    var gqlRequest = JsonSerializer.Deserialize<GraphQLRequest>(jsonBytes.AsSpan(), _serializerOptions);
+                    return (true, gqlRequest, null);
+
+                case JsonTokenType.StartArray:
+                    var gqlRequests = JsonSerializer.Deserialize<GraphQLRequest[]>(jsonBytes.AsSpan(), _serializerOptions);
+                    return (true, null, gqlRequests);
+
+                default:
+                    return (false, null, null); // BadRequest
+            }
+        }
+
+        private static JsonTokenType GetTokenType(byte[] bytes)
+        {
+            var reader = new Utf8JsonReader(bytes.AsSpan());
+            reader.Read();
```

Comment:
Probably if this `Read` call returns `false` there will be nothing bad.

Reply:
I thought that too looking at the docs, but with the unit test we had giving it `0` in the body, it threw. We can revisit this once I've finished moving things around. I'll probably handle the false too.

Reply:
Or maybe handle `Content-Length` = 0 earlier for `POST` requests